### ISSUE TITLE
Show zoom level for the active page

### ISSUE THREAD

**Issue by zalexzperez** [eez-open/studio#563]
When you start working on multiple pages, it's normal that they end up having different zoom values, especially on small displays.

If you want to visually compare widget positions/sizes between different screens, you need a common zoom value, so having a permanent field showing the current page zoom is of help.

And maybe, a button to reset zoom for all the project pages at once would be a good addition as well.

**Comment by mvladic** [comment]
You can double click on empty space to reset zoom and center the screen.

**Comment by zalexzperez** [comment]
That's true for the second feature and I actually do that all the time, but I still have 10 "disorganized" screens. I will have to get used to double click before changing to another page. Although I constantly need to zoom in to work on my low-resolution display. Perhaps a general zoom level for all the pages would be better than a global reset zoom button. Like some design programs:
![image](https://github.com/user-attachments/assets/a610adb0-6e01-4e18-afec-18c76e5ee7cd)

**Comment by mvladic** [comment]
Makes sense. I'm not reading carefully again ;)

**Comment by mvladic** [comment]
I'm not sure where to put this zoom level control and how user will choose whether to set the zoom level for the selected page or globally for all pages.

Alternatively, what you want can be done like this:

![same zoom](https://github.com/user-attachments/assets/ba799c7f-b5cf-4177-a654-22afd4e899df)

What do you think?

**Comment by zalexzperez** [comment]
That looks very nice.

Now, I hope you're able to read all of this :P

About the zoom level control, I initially took inspiration in CorelDRAW. This is how it works:

https://github.com/user-attachments/assets/832a8a54-cf96-495c-b0a5-126df2010496

As you can see, both the zoom level and the position are global. 

To be honest, since this feature was about enhancing zoom control to be able to compare widget positions between screens more easily and also about not having a mess of different zoom levels between all the pages, I think it would be best to have a global zoom by default.

By having it like this, user only has to take care of zoom from a single place (page).

About having a global position control, it has its own charm: again, if we want to compare two pages, having the same exact position is helpful. But I can also see an important disadvantage: e.g. you zoom all the way into a remote area of flow actions and go to another page, you will likely end up viewing an empty space. Nothing a quick zoom out can't fix, though.

------------------

So,  I'd do it like this:

1- A current zoom level dropdown with a few predefined values.
2- A button to enable/disable global zoom.

When global zoom is disabled, everything can be as it is now, with the new context menu showing Reset and Set options.
When global zoom button is enabled, all the pages automatically are set to the current's page zoom level. Context menu will still work, but the Set option shouldn't be available for use.

The global position control with an extra button... although nice to have, I think it's a bit overkill.

And finally, about where to place the controls, this would be my preferred location:

![image](https://github.com/user-attachments/assets/77b9aeb1-6ef5-487c-aa1f-04f1b0adf517)

**Comment by mvladic** [comment]
I added this dropdown:

![image](https://github.com/user-attachments/assets/52374a6b-4a63-4e43-bd4c-e4f077027297)

Right click on the empty area when Global Zoom is enabled shows this context menu:

![image](https://github.com/user-attachments/assets/967371f9-1c1c-4d20-b9d9-b9d29dbdee6e)

If Global Zoom is not enabled, it shows:

![image](https://github.com/user-attachments/assets/dbd1bebc-45e7-4d2b-8362-594351da89b0)

"Center view" will only put page in the center, it will not change zoom.
Set the Same Zoom for All Pages" will only affect the zoom, not position.

Doubleclick on the page both center view and sets the zoom to 100%, as before.

**Comment by zalexzperez** [comment]
Great job! Love this feature!

It is also worth mentioning that it is possible to quickly center view on all pages as well:

![image](https://github.com/user-attachments/assets/ad33578c-e897-4878-b704-f21d03cb81c3)
